tests/test_solver.py

```diff
@@ -14,15 +14,38 @@ def tmpcache():
             yield
 
 
+def _edit(v):
+    return (v.tolist() if isinstance(v, numpy.ndarray)
+        else tuple(map(_edit, v)) if isinstance(v, tuple)
+        else {k: _edit(v) for (k, v) in v.items()} if isinstance(v, dict)
+        else v) # convert arrays to lists so we can use assertEqual
+
+
 def _test_recursion_cache(testcase, solver_iter):
-    edit = lambda v: v.tolist() if numeric.isarray(v) else tuple(map(edit, v)) if isinstance(v, tuple) else v  # convert arrays to lists
-    read = lambda n: tuple(edit(item) for i, item in zip(range(n), solver_iter()))
+    read = lambda n: tuple(_edit(item) for i, item in zip(range(n), solver_iter))
     reference = read(5)
-    for lengths in [1, 2, 3], [1, 3, 2], [0, 3, 5]:
+    for lengths in [1, 2, 3], [1, 3, 2], [3, 5]:
```

Comment:
Why is `0` removed from the last set of lengths?

Reply:
restored, along with lambda functions